crates/assistant_tool/src/assistant_tool.rs

```diff
@@ -36,4 +45,47 @@ pub trait Tool: 'static + Send + Sync {
         window: &mut Window,
         cx: &mut App,
     ) -> Task<Result<String>>;
+
+    /// Renders the tool's input when the user expands it.
+    fn render_input(
+        self: Arc<Self>,
+        input: serde_json::Value,
+        _lua_language: Option<Arc<Language>>,
```

Comment:
I think the scripting tool should be loading the Lua language from the language registry instead of having it passed in here.

Having it be on the `Tool` trait is leaking details about the scripting tool implementation.